src/modbus_rtu_client.cpp

```diff
@@ -2,14 +2,226 @@
+    } value_8;
+} ;
+
+#endif
+
+
+DataVectorUint8 ModbusRTUClient::response_without_protocol_data( const DataVectorUint8& raw_response, std::size_t payload_length ) {
+    // strip address and function bytes, but include bytecount byte
+    const int offset_protocol_bytes { 3 };
+    return DataVectorUint8( raw_response.cbegin() +
+                            offset_protocol_bytes,
+                            raw_response.cbegin() +
+                            offset_protocol_bytes +
+                            payload_length  );
+
+}
+
+const DataVectorUint8 ModbusRTUClient::read_holding_register(uint8_t unit_id,
+                                                             uint16_t first_register_address,
+                                                             uint16_t num_registers_to_read,
+                                                             bool return_only_registers_bytes ) const {
+
+    using namespace std::string_literals;
+
+    #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+
+    if ( num_registers_to_read > everest::modbus::consts::rtu::MAX_REGISTER_PER_MESSAGE )
+        throw everest::modbus::exceptions::message_size_exception( ""s + __PRETTY_FUNCTION__ +
+                                                                   " Requested number of 16 bit registers " + std::to_string( num_registers_to_read ) +
+                                                                   " would exceed allowed message size of " + std::to_string( everest::modbus::consts::rtu::MAX_REGISTER_PER_MESSAGE ) + " registers " );
+
+    DataVectorUint8 body = utils::build_read_holding_register_message_body(first_register_address, num_registers_to_read);
+
+    DataVectorUint8 full_message = full_message_from_body(body, consts::READ_HOLDING_REGISTER_MESSAGE_LENGTH, unit_id );
+
+    conn.send_bytes(full_message);
+    modbus::DataVectorUint8 response = conn.receive_bytes( max_adu_size() );
+    uint16_t payload_size = validate_response( response, full_message ) ;
+    return return_only_registers_bytes ? response_without_protocol_data( response, payload_size ) : response;
+
+    #else
+
+    static_assert( false , "implementation currently done for little endian only" );
+
+    #endif
+
+}
+
+DataVectorUint8 ModbusDataContainerUint16::get_payload_as_bigendian() const {
+
+    DataVectorUint8 result;
+    result.reserve( m_payload.size() * sizeof( DataVectorUint16::value_type ));
+
+    if ( m_byte_order == ByteOrder::LittleEndian )
+        for( uint16_t value : m_payload ) {
+            SwapIt s { value };
+            result.push_back( s.value_8.hi   );
+            result.push_back( s.value_8.low  );
+        }
+    else
+        for( uint16_t value : m_payload ) {
+            SwapIt s { value };
+            result.push_back( s.value_8.low );
+            result.push_back( s.value_8.hi  );
+        }
+
+    return result;
+}
+
+DataVectorUint8 everest::modbus::ModbusRTUClient::write_multiple_regiser( uint8_t unit_id,
```

Comment:
Well... this got disimproved...
```suggestion
DataVectorUint8 everest::modbus::ModbusRTUClient::write_multiple_registers( uint8_t unit_id,
```

Reply:
Also, please check the alignment of the parameters in the following lines. On my machine they seem to no longer match up to the first parameter indentation.

Reply:
This has been resolved! Many thanks!